src/components/compound/DlInput/DlTextInput.vue

```diff
@@ -590,8 +590,14 @@ export default defineComponent({
         },
         onChange(e: any): void {
             this.isMenuOpen = true
-            this.$emit('input', e.target.value, e)
-            this.$emit('update:model-value', e.target.value)
+            const trimmedValue = e.target.value.trim()
+            if (trimmedValue !== e.target.value) {
```

Comment:
This condition is not necessary, as you can jut always send the trimmed value because it's equals to the original value